server/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/InternalScriptedMetric.java

```diff
@@ -89,15 +89,16 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
         InternalScriptedMetric firstAggregation = ((InternalScriptedMetric) aggregations.get(0));
         List<Object> aggregation;
         if (firstAggregation.reduceScript != null && reduceContext.isFinalReduce()) {
-            Map<String, Object> vars = new HashMap<>();
-            vars.put("_aggs", aggregationObjects);
+            Map<String, Object> params = new HashMap<>();
             if (firstAggregation.reduceScript.getParams() != null) {
-                vars.putAll(firstAggregation.reduceScript.getParams());
+                params.putAll(firstAggregation.reduceScript.getParams());
             }
-            ExecutableScript.Factory factory = reduceContext.scriptService().compile(
-                firstAggregation.reduceScript, ExecutableScript.AGGS_CONTEXT);
-            ExecutableScript script = factory.newInstance(vars);
-            aggregation = Collections.singletonList(script.run());
+            params.put("_aggs", aggregationObjects);
```

Comment:
Is this needed? I thought that the reason to use script contexts was to that we then wouldn't need to put the `_aggs` variable in the params because it is passed directly to the script?

Reply:
This (along with analogous code for `_agg` in ScriptedMetricAggregatorFactory.java and some of the integration tests I did not modify) is needed if we want to retain backwards compatibility with the old way for one ES version, letting people migrate to the context variables while both still work.

If a clean break is acceptable then this would be removed and some of the tests could be cleaned up. But I would assume we want some compatibility leniency and possibly some warning messages or other notification (which I have not implemented yet -- item 2 on my list in the first comment on this PR).

Reply:
Ok, understood. I'm +1 on keeping this for backwards compatibility, thanks for explaining. Maybe we could add a comment just noting that its there for backwards compatibility to avoid someone removing it thinking its just left over?

Reply:
Good call. I will add a comment here and in other relevant spots.

Reply:
Glad you brought this up -- when looking for spots that needed a comment I found a leftover usage of `params.get("_agg")` in ScriptedMetricAggregator.java that needed to be replaced with a direct usage of the agg object. No functional issue but it would've had to get cleaned up later.